osbuild/api.py

```diff
@@ -57,6 +56,11 @@ def _make_socket_dir(cls):
         """Called to create the temporary socket dir"""
         return tempfile.TemporaryDirectory(prefix="api-", dir="/run/osbuild")
 
+    def _dispatch(self, sock: jsoncomm.Socket):
+        """Called when data is available on the socket"""
+        msg, fds, addr = sock.recv()
+        self._message(msg, fds, sock, addr)
```

Comment:
Maybe even worth it to call `fds.close()` here, and make any message dispatcher steal the FDs from the `FdSet`?

Reply:
I like the idea!